test/general/test_items.py

```diff
@@ -8,12 +8,34 @@ class TestBase(unittest.TestCase):
     def test_create_item(self):
         """Test that a world can successfully create all items in its datapackage"""
         for game_name, world_type in AutoWorldRegister.world_types.items():
-            proxy_world = setup_solo_multiworld(world_type, ()).worlds[1]
+            multiworld = setup_solo_multiworld(world_type, steps=("generate_early", "create_regions", "create_items"))
+            proxy_world = multiworld.worlds[1]
+            empty_prog_items = multiworld.state.prog_items.copy()
             for item_name in world_type.item_name_to_id:
                 with self.subTest("Create Item", item_name=item_name, game_name=game_name):
                     item = proxy_world.create_item(item_name)
+
+                with self.subTest("Item Name", item_name=item_name, game_name=game_name):
                     self.assertEqual(item.name, item_name)
 
+                if item.advancement:
+                    with self.subTest("Item State Collect", item_name=item_name, game_name=game_name):
+                        multiworld.state.collect(item, True)
+
+                    with self.subTest("Item State Remove", item_name=item_name, game_name=game_name):
+                        multiworld.state.remove(item)
+
+                        self.assertEqual(multiworld.state.prog_items, empty_prog_items,
+                                         "Item Collect -> Remove should restore empty state.")
+                else:
+                    with self.subTest("Item State Collect No Change", item_name=item_name, game_name=game_name):
+                        # Non-Advancement should not modify state.
+                        base_state = multiworld.state.prog_items.copy()
+                        multiworld.state.collect(item)
+                        self.assertEqual(base_state, multiworld.state.prog_items)
+
+                multiworld.state.prog_items = empty_prog_items
```

Comment:
per discussion in ap-core-dev, this line is setting by reference and causing errors to go unnoticed

a couple of things seem to work, that I don't have much of an opinion on which is better
* deepcopy when copying prog_items
* copying CollectionState instead and using that state's prog_items when comparing
* copy a new CollectionState for each item_name and using that state's collect() instead of copying the state back to multiworld each loop